src/include/test/unit_timetravel.cc

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file   unit_timetravel.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Test time traveling support in TileDB-Vector-Search
+ *
+ */
+
+#include <algorithm>
+#include <catch2/catch_all.hpp>
+#include <chrono>
```

Comment:
Do we need this file?

Reply:
For this PR, no.  For TileDB-Vector-Search, absolutely.  And it's well overdue.  We have been incorporating time-travel bits into the C++ library, but  have not unit tests for it -- partly because there isn't any spec or other explanation of correct behavior and thus no way to write actual unit tests.

But if unit tests ever get written for time traveling -- and I hope they will -- this is the file where they should go.